make/Build.java

```diff
@@ -0,0 +1,1617 @@
+        public Map<String, String> getMakeArgs() {
+            var jars = List.of(jar, jcommander.jar, guice.jar);
+            return Map.of(
+                    "TESTNG_JARS", jars.stream()
+                            .map(Path::toString)
+                            .collect(Collectors.joining(" ")),
+                    "TESTNG_NOTICES", license.toString());
+        }
+    }
+
+    static class Guice extends Dependency {
+        Path jar;
+
+        public Guice(Config config, Tools tools) {
+            super("google_guice", config, tools);
+        }
+
+        @Override
+        public void setup() throws Fault {
+            jar = config.getPath("GUICE_JAR");
+            if (jar == null) {
+                jar = downloadStandardJar(((urlBase, version) ->
+                        urlBase
+                                + "/com/google/inject/guice/"
+                                + version
+                                + "/guice-" + version + ".jar"
+                ));
+            }
+        }
+
+        @Override
+        public void verify() throws Fault {
+            checkFile(jar);
+        }
+    }
+
+    static class JCommander extends Dependency {
+        Path jar;
+
+        public JCommander(Config config, Tools tools) {
+            super("jcommander", config, tools);
+        }
+
+        @Override
+        public void setup() throws Fault {
+            jar = config.getPath("JCOMMANDER_JAR");
+            if (jar == null) {
+                jar = downloadStandardJar(((urlBase, version) ->
+                        urlBase
+                                + "/com/beust/jcommander/"
+                                + version
+                                + "/jcommander-" + version + ".jar"
+                ));
+            }
+        }
+
+        @Override
+        public void verify() throws Fault {
+            checkFile(jar);
+        }
+    }
+
+    /**
+     * Generates a script to run "make", based on the set of dependencies.
+     */
+    static class MakeScript {
+        private final Config config;
+        MakeScript(Config config) {
+            this.config = config;
+        }
+
+        void writeFile(Path file, List<? extends Dependency> deps) throws Fault {
+            var allMakeArgs = new TreeMap<String, String>();
+            deps.forEach(d -> allMakeArgs.putAll(d.getMakeArgs()));
+
+            try (PrintWriter out = new PrintWriter(Files.newBufferedWriter(file))) {
+                out.println("#!/bin/sh");
+                out.println();
+                out.println("cd \"" + config.rootDir.resolve("make") + "\"");
+                out.println("make \\");
+                allMakeArgs.forEach((name, value) ->
+                        out.printf("    %s=\"%s\" \\%n", name, value));
+                out.println("    \"$@\"");
+            } catch (IOException e) {
+                throw new Fault("Error writing make command script: " + file + ": " + e);
+            }
+        }
+    }
+}
```

Comment:
Why isn't jcheck complaining about the lack of newline at EOF? I thought it should do that.